src/models/sqlOutputContentProvider.ts

```diff
@@ -237,53 +94,77 @@ export class SqlOutputContentProvider {
             : this._queryResultsMap.get(uri).queryRunner.isExecutingQuery;
     }
 
-    public runQuery(
+    public async runQuery(
             statusView: any, uri: string,
-            selection: ISelectionData, title: string): void {
+            selection: ISelectionData, title: string): Promise<void> {
         // execute the query with a query runner
-        this.runQueryCallback(statusView, uri, selection, title,
+        await this.runQueryCallback(statusView ? statusView : this._statusView, uri, title,
             (queryRunner) => {
                 if (queryRunner) {
                     queryRunner.runQuery(selection);
                 }
             });
     }
 
-    public runCurrentStatement(
+    public async runCurrentStatement(
             statusView: any, uri: string,
-            selection: ISelectionData, title: string): void {
+            selection: ISelectionData, title: string): Promise<void> {
         // execute the statement with a query runner
-        this.runQueryCallback(statusView, uri, selection, title,
+        await this.runQueryCallback(statusView ? statusView : this._statusView, uri, title,
             (queryRunner) => {
                 if (queryRunner) {
                     queryRunner.runStatement(selection.startLine, selection.startColumn);
                 }
             });
     }
 
-    private runQueryCallback(
-            statusView: any, uri: string,
-            selection: ISelectionData, title: string,
-            queryCallback: any): void {
-
-        let resultsUri = this.getResultsUri(uri);
-        let queryRunner = this.createQueryRunner(statusView, uri, resultsUri, selection, title);
+    private async runQueryCallback(
+            statusView: any, uri: string, title: string,
+            queryCallback: any): Promise<void> {
+        let queryRunner = this.createQueryRunner(statusView ? statusView : this._statusView, uri, title);
+        if (this._panels.has(uri)) {
+            let panelController = this._panels.get(uri);
+            if (panelController.isDisposed) {
+                this._panels.delete(uri);
+                await this.createWebviewController(uri, title, queryRunner);
+            } else {
+                return queryCallback(queryRunner);
+            }
```

Comment:
This could result in something being returned but the signature says it's a void return - do you expect the callback to have a return type? Probably best just to have it also be void and then call it directly - then have the return called separately.

```
queryCallback(queryRunner);
return;
```

Reply:
It returns a void promise. Are you looking at the old signature?

Reply:
I'm saying the callback itself could return anything since you define it as an any

queryCallback: any

I would suggest updating it to specify that it should have a void return type :

```
queryCallback: (queryRunner: QueryRunner) => void
```

---
In reply to: [322361648](https://github.com/microsoft/vscode-mssql/pull/1282#discussion_r322361648) [](ancestors = 322361648)

Reply:
And that you shouldn't be returning the result of that call anyways since this method returns Promise<void> (so just calling return will be essentially the same as return Promise.resolve())

---
In reply to: [322382850](https://github.com/microsoft/vscode-mssql/pull/1282#discussion_r322382850) [](ancestors = 322382850,322361648)